pkg/codegen/python/gen.go

```diff
@@ -250,21 +250,30 @@ func (mod *modContext) objectType(t *schema.ObjectType, input bool) string {
 	return fmt.Sprintf("'%s%s%s'", modName, prefix, name)
 }
 
-func (mod *modContext) tokenToEnum(tok string) string {
-	modName, name := mod.tokenToModule(tok), tokenToName(tok)
+func (mod *modContext) enumType(enum *schema.EnumType) string {
+	tok := enum.Token
+	pkgName, modName, name := enum.PackageReference.Name(), mod.tokenToModule(tok), tokenToName(tok)
 
-	if modName == "" && modName != mod.mod {
-		return fmt.Sprintf("'_root_enums.%s'", name)
-	}
+	if pkgName == mod.pkg.Name() {
+		if modName == "" && modName != mod.mod {
+			return fmt.Sprintf("'_root_enums.%s'", name)
+		}
 
-	if modName == mod.mod {
-		modName = ""
+		if modName == mod.mod {
+			modName = ""
+		}
+		if modName != "" {
+			modName = strings.ReplaceAll(modName, "/", ".") + "."
+		}
+
+		return fmt.Sprintf("'%s%s'", modName, name)
 	}
+
+	// Handle external enum type.
 	if modName != "" {
-		modName = "_" + strings.ReplaceAll(modName, "/", ".") + "."
+		modName = strings.ReplaceAll(modName, "/", ".") + "."
```

Comment:
optional: I think we could maybe avoid doing the same transformations in two places and treat that if block more as a special case that just modifies pkgName and modName:

```go
if ... {
  pkgName = "" // local reference
  // ...
}

if modName != "" {
  modName = strings.ReplaceAll(...)
}

// And then something concatenation based, or like the following.
var parts []string
if pkgName != "" { parts = append(parts, pyPack(pkgName)) }
if modName != "" { parts = append(parts, strings.ReplaceAll(modName, ...)) }
parts = append(parts, name)
return strings.Join(parts, ".")
```

That reads as two phases: extract the necessary information, then combine it to produce the result.